### PR TITLE
I create baseDao class and taskDao which extends from it.

### PR DESCRIPTION
**Pull request by Flozik**
I create class BaseDao.
TaskDao extend from BaseDao
And I clear TaskDaoImpl

### CODE REVIEW

File: src/main/java/com/jc/tm/database/dao/BaseDao.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    void update(E e);
+
+    //get
+    E getById(int id);
+
+    //get all
+    List<E> getAll();
+
+    //delete
+    void delete(E e);
```
Comment (komarichyn) [inline src/main/java/com/jc/tm/database/dao/BaseDao.java:22]: everything ok, good job. but lets add return type for delete. it is usefull for future. if we will remove some object, but object not exists into database we can return null and according null we can recognize ttaht object doest nott exist and display correct error

File: src/main/java/com/jc/tm/database/dao/TaskDao.java [hunk 2]
```diff
@@ -2,20 +2,31 @@
...
 import com.jc.tm.database.entity.Task;
 
-public interface TaskDao {
+import java.util.List;
 
-    //create or insert task
-    void insetTask(Task task);
+public interface TaskDao extends BaseDao<Task> {
+    @Override
+    default void insert(Task task) {
```
Comment (komarichyn) [inline src/main/java/com/jc/tm/database/dao/TaskDao.java:9]: do we need defaultt implementaiton of interface?

**Review by komarichyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/jc/tm/database/dao/BaseDao.java (hunk 1 above)
Comment (Flozik) [reply src/main/java/com/jc/tm/database/dao/BaseDao.java]: delete method was changed from void to boolean.

File: src/main/java/com/jc/tm/database/dao/TaskDao.java (hunk 2 above)
Comment (Flozik) [reply src/main/java/com/jc/tm/database/dao/TaskDao.java]: Default was deleted.

**Review by komarichyn — APPROVED** [review]

**Pull request merged**